capitals.rb

```diff
@@ -150,3 +150,57 @@
     name: "Wyoming",
     capital: "Cheyenne"
 }]
+
+puts "Let's test your skills...You ready? Y/N"
+ready = gets.chomp
+states.shuffle!
+states.each do |state|
+  state[:times_correct] = 0
+  state[:times_wrong] = 0
+end
+i = 0
+
+if ready == "Y"
+    while i < states.length
+      puts "You better tell me the capital of #{states[i][:name]}"
+      user_answer = gets.chomp
+      if user_answer == states[i][:capital]
+        states[i][:times_correct] += 1
+        puts "Good job, you are #{states[i][:times_correct]} for " + (states[i][:times_wrong] + states[i][:times_correct]).to_s + " on this state."
+      else
+        states[i][:times_wrong] += 1
+        puts "Wrong answer you are #{states[i][:times_correct]} for " + (states[i][:times_wrong] + states[i][:times_correct]).to_s + " on this state."
+      end
+      i += 1
+    end
+    i = 0
+else
+  puts "Okay. Have a good day"
+end
+
+
+puts "Want to play again? Y/N"
+ready = gets.chomp
+
+while ready == "Y"
+    states.sort_by! {|state| state[:times_wrong]}.reverse!
```

Comment:
another way to do this is to sort by number correct so you don't have to add reverse